### PR TITLE
Makefile cleanup

### PR DESCRIPTION
**Pull request by dsroberts**
Revamp makefiles. make check now runs appropriate tests wherever its invoked. Top level behaviour remains the same, make check in demos only tests demos, make check in an individual test only tests that, etc. Most make check invocations depend on test output. Some tests were modified to make this work properly. make clean now removes all artefacts from testing.

### CODE REVIEW

File: tests/analytical_comparisons/analytical.py [hunk 1]
```diff
@@ -72,7 +73,8 @@
 
 
 def get_case(cases, config):
-    config = config.split("/")
+    print(config)
```
Comment (angus-g) [inline tests/analytical_comparisons/analytical.py:76]:
```suggestion
```

Drop?

**Review by angus-g — COMMENTED** [review]
Awesome, thanks! Should we use `make check` in https://github.com/g-adopt/g-adopt/blob/1d10b20dbbfbc1cd93b7fd51e3b27589b2c7e7bb/.github/workflows/test.yml#L113? Although it would need to learn how to do the JUnit XML output.

File: tests/analytical_comparisons/analytical.py (hunk 1 above)
Comment (dsroberts) [reply tests/analytical_comparisons/analytical.py:76]: Yes please. Looks like I left some debugging in...

**Comment by dsroberts** [comment]
I think the actions workflows are fine the way they are. I envisioned that this setup would be used for interactive testing during development or deployment to a new system. The machines don't need anything simplified for them...

**Review by angus-g — APPROVED** [review]
All good then!

**Pull request merged**